### PR TITLE
EUI-3753-blank-pba-not-discarded

### PR DESCRIPTION
**Pull request by RiteshHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3753

### Change description ###
Blank PBA should be discarded

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/register/containers/register/register.component.ts [hunk 1]
```diff
@@ -141,6 +141,15 @@ export class RegisterComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   public onPageContinue(formDraft: any): void {
+    if (formDraft && formDraft['controls']) {
+      if (Object.keys(formDraft.controls).filter(value => value.startsWith('PBANumber')).length > 1) {
+        Object.keys(formDraft.controls).filter(value => value.startsWith('PBANumber')).forEach(key => {
+          if (formDraft['controls'][key].value === null) {
+            this.store.dispatch(new fromStore.RemovePBANumber('remove' + key));
```
Comment (rayliangatos) [inline src/register/containers/register/register.component.ts:148]: there will be race condition on this.store.dispatch(new fromStore.RemovePBANumber('remove' + key));
and 
this.store.dispatch(new fromStore.SaveFormData({value, pageId: this.pageId}));

And it's not necessary to do two dispatches in one method, the best place to put this logic should be in 
this.store.dispatch(new fromStore.SaveFormData({value, pageId: this.pageId}));
and filter out those value is null/empty when saving them.

Comment (RiteshHMCTS) [reply src/register/containers/register/register.component.ts:148]: @rayliangatos I was thinking of to implement this logic, in fact did it but issue is unless we save the data we get those PBA element which has no values which is not what expected.

File: test/e2e/features/step_definitions/createOrganisation.steps.js [hunk 2]
```diff
@@ -76,8 +76,8 @@ defineSupportCode(function ({ Given, When, Then }) {
     browser.sleep(MID_DELAY);
     await createOrganisationObject.PBANumber1.isDisplayed();
     await createOrganisationObject.enterPBANumber();
-    await createOrganisationObject.PBANumber2.isDisplayed();
-    await createOrganisationObject.enterPBA2Number();
+    // await createOrganisationObject.PBANumber2.isDisplayed();
```
Comment (johnbenjamin-hmcts) [inline test/e2e/features/step_definitions/createOrganisation.steps.js:79]: @RiteshHMCTS I think you can delete the commented statements.

File: test/accessibility/tests/unauthenticatedUrls.js [hunk 3]
```diff
@@ -83,7 +83,7 @@ describe('Pa11y tests', function () {
         actions.push(...PallyActions.clickElement(continueButtonLocator));
 
         actions.push(...PallyActions.inputField('#PBANumber1', 'PBA1234567'));
-        actions.push(...PallyActions.inputField('#PBANumber2', 'PBA7654321'));
+        // actions.push(...PallyActions.inputField('#PBANumber2', 'PBA7654321'));
```
Comment (johnbenjamin-hmcts) [inline test/accessibility/tests/unauthenticatedUrls.js:86]: @RiteshHMCTS I think you can delete the commented statement.

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Comment by paulhowes-HMCTS** [comment]
This PR and corresponding branch has been merged into the new Multiple PBA Feature branch
New branch: https://github.com/hmcts/rpx-xui-manage-organisations/tree/feature/EUI-3087-3088-multiple-pbas
New PR: https://github.com/hmcts/rpx-xui-manage-organisations/pull/760
DO NOT REOPEN THIS PR!

**Pull request closed without merging**